controllers/storagecluster/external_resources.go

```diff
@@ -124,18 +124,17 @@ func (r *StorageClusterReconciler) externalSecretDataChecksum(instance *ocsv1.St
 	return sha512sum(found.Data[externalClusterDetailsKey])
 }
 
-func (r *StorageClusterReconciler) sameExternalSecretData(instance *ocsv1.StorageCluster) bool {
+func (r *StorageClusterReconciler) sameExternalSecretData(instance *ocsv1.StorageCluster) (string, bool) {
```

Comment:
Let's not return the hash from this function, The function name suggests it should return a boolean so let it be that way.